tdrs-backend/tdpservice/users/api/authorization_check.py

```diff
@@ -14,23 +17,24 @@ class AuthorizationCheck(APIView):
 
     query_string = False
     pattern_name = "authorization-check"
+    permission_classes = [AllowAny]
```

Comment:
Wondering why this is the right permission class to use here — what is the benefit to allowing unauthenticated users to query this endpoint? 

https://www.django-rest-framework.org/api-guide/permissions/#allowany

Reply:
It was quite a while ago, so I'm sure you don't remember but we discussed this at length with you and Ryan when the backend change was made. The endpoint exists to check if a user is authorized. If the endpoint is restricted it will return an error instead of the desired response.

Reply:
Gotcha! Yes, I remember our conversation but I found myself wondering the same thing re-reading the code this time around. Maybe an opportunity for a code comment or other documentation? 

I'm trying to think through why we wouldn't want to use the `IsAuthenticated` permission in the Django Rest Framework here — maybe there's a detail or use case I'm not considering.

With the current code (`AllowAny`), when I hit the endpoint as an unauthenticated user, I get this response:

<img width="339" alt="Screen Shot 2020-11-13 at 9 29 29 AM" src="https://user-images.githubusercontent.com/3209501/99090584-71d36900-2594-11eb-8bd7-0303191377d8.png">

```
{
    "authenticated": false
}
```

I don't get any user data, which is approriate. The response is served up by custom code in `authorization_check.py`.  

If we used `IsAuthenticated`, a non-logged-in user would get this response:

## 

<img width="526" alt="Screen Shot 2020-11-13 at 9 31 12 AM" src="https://user-images.githubusercontent.com/3209501/99090587-726bff80-2594-11eb-95b2-eea42f548f7d.png">

```
{
    "detail": "Authentication credentials were not provided."
}
```

No custom code is called here; Django REST framework provides the response out of the box, which reduces the surface for potential bugs. 

We'd also get the idiomatic HTTP 403 Forbidden Code. This may have caching benefits — [200 OK responses are cacheable by default](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/200), do we want the browser to cache an "200 OK authentication: false" response?

I may be missing use cases or relevant details here! Either way I think a bit more documentation on this choice will be helpful to future devs reading the code.

Reply:
Technically it could be done either way. In my view it is generally bad practice to send a user where you can expect a possible error. Of course the front end could handle it, as it has to handle any `403`. If you feel strongly about it, I suppose we could put in an issue to adjust how that works. It wouldn't be a heavy lift, but more than I would want to do to hold this up. Thoughts?

Reply:
@carltonsmith I don't feel too strongly about it, especially since this is a simple GET request for authentication and doesn't govern any sensitive actions or data. Thanks for the explanation!

Reply:
It's my pleasure